packages/server/src/threads/query.ts

```diff
@@ -170,7 +170,12 @@ class QueryRunner {
     }
 
     // get all the potential fields in the schema
-    let keys = rows.flatMap(Object.keys)
```

Comment:
You could just wrap this into a new Set, avoiding the foreach:
`let keys = new Set(rows.flatMap(Object.keys))`